### PR TITLE
remove support for 1.6 LTS

### PR DESCRIPTION
**Pull request by storopoli**
This removes supports for Julia 1.6 LTS and updates the `CI.yml` to use the latest Julia stable release.

### CODE REVIEW

**Comment by rikhuijzer** [comment]
> This removes supports for Julia 1.6 LTS and updates the CI.yml to use the latest Julia stable release.

Agreed. Ain't nobody got time for that.

To get rid of those "Julia CI (1.6, [...]) Expected --- Waiting for status to be reported", remove the checks from the branch protection rules in the settings (Settings -> Branches -> Branch Protection Rules -> main -> Edit and modify the "Status checks that are required." area there.

**Comment by rikhuijzer** [comment]
Maybe also put it in the `Project.toml`? Does it work on 1.7? Then make that the minimum version in the `Project.toml`.

**Comment by rikhuijzer** [comment]
And also write it somewhere in the top of the documentation what is the least required Julia version. Otherwise, people will be confused when they get an old version of `TuringGLM` while they should have a new version

**Comment by storopoli** [comment]
> Maybe also put it in the `Project.toml`? Does it work on 1.7? Then make that the minimum version in the `Project.toml`.

Yes it does work on Julia 1.7.

> And also write it somewhere in the top of the documentation what is the least required Julia version. Otherwise, people will be confused when they get an old version of `TuringGLM` while they should have a new version

Sure!

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -21,8 +21,7 @@ jobs:
       fail-fast: false
       matrix:
         version:
-          - '1.6'
-          - '1.8'
+          - '1'
```
Comment (rikhuijzer) [inline .github/workflows/CI.yml:24]:
```suggestion
          - '1.7'
          - '1'
```

Comment (rikhuijzer) [reply .github/workflows/CI.yml:24]: To keep an eye on whether 1.7 keeps working

**Comment by storopoli** [comment]
Ok, approve it and we can enable auto-merge.

**Review by rikhuijzer — APPROVED** [review]

**Pull request merged**